### PR TITLE
Deal with held messages, deal with empty queue

### PR DESCRIPTION
**Pull request by bernhardschmidt**
Thanks for your work, it is immensely useful. I've added a few small fixes, see below.

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
Looks great! I've added you as a committer, please feel free to merge this

**Comment by antoniovl** [comment]
Any plans to merge this PR anytime soon? :-)

**Comment by bernhardschmidt** [comment]
Hrm, I cannot merge it.

@alexjurkiewicz Could you check the permissions again and/or merge yourself?

File: apq.py [hunk 1]
```diff
@@ -38,12 +38,15 @@ def parse_mq(args):
     for line in mailq_stdout.splitlines():
         if not line or line[:10] == '-Queue ID-' or line[:2] == '--':
             continue
-        if line[0] in '0123456789ABCDEF':
+        if line[0] in '0123456789ABCDEF': # XXX: unsure whether long queue IDs also start like that
```
Comment (alexjurkiewicz) [inline apq.py:41]: Is this comment still valid?

**Review by alexjurkiewicz — CHANGES_REQUESTED** [review]
I've had a bit of time to review this PR. Please have a look and let me know.

BTW, once this PR is merged, I plan to archive this repo, so if you prefer feel free to close this and pre-emptively fork.

File: apq.py [hunk 2]
```diff
@@ -57,6 +60,8 @@ def parse_mq(args):
             msgs[curmsg]['recipient'] = line.strip()
         elif line.lstrip(' ')[0] == '(':
             msgs[curmsg]['reason'] = line.strip()[1:-1].replace('\n', ' ')
+        elif line == 'Mail queue is empty':
```
Comment (alexjurkiewicz) [inline apq.py:63]: This should be part of the test on line 39.

**Pull request closed without merging**